src/main/java/com/google/sps/servlets/EventsServlet.java

```diff
@@ -211,4 +259,47 @@ public String sanitizeInput(String input) throws java.io.UnsupportedEncodingExce
       input = new String( input.getBytes("UTF-8") , 0, Math.min(MAX_STRING_BYTES, input.length()), "UTF-8");
       return input;
   }
+
+  public long getEventId() {
+      Query query = new Query(EVENT_NUM);
+      PreparedQuery pq = datastore.prepare(query);
+      QueryResultList<Entity> eventNumResult = pq.asQueryResultList(FetchOptions.Builder.withDefaults());
+
+      Entity eventNumEntity;
+      long eventNumValue;
+      if (eventNumResult.size() == 0) {
+          eventNumEntity = new Entity(EVENT_NUM);
+          eventNumValue = 0;
+          eventNumEntity.setProperty(EVENT_NUM_VALUE, eventNumValue);
+          datastore.put(eventNumEntity);      
+      }
+      else {
+          eventNumEntity = eventNumResult.get(0);
+          eventNumValue = (long)eventNumEntity.getProperty(EVENT_NUM_VALUE)+1;
+          eventNumEntity.setProperty(EVENT_NUM_VALUE, eventNumValue);
+          datastore.put(eventNumEntity);
+      }
+
+      return eventNumValue;
+  }
+
+  public void updateUserCreatedEvents(String userId, long eventId) {
+      Query userQuery = new Query(User.DATA_TYPE).setFilter(new FilterPredicate(User.ID, FilterOperator.EQUAL, userId));
+      Entity entity = datastore.prepare(userQuery).asSingleEntity();
+      ArrayList<Long> createdEvents =(ArrayList<Long>) entity.getProperty(User.CREATED_EVENTS);
+      if (createdEvents == null) createdEvents = new ArrayList<Long>();
+      createdEvents.add(eventId);
+      entity.setProperty(User.CREATED_EVENTS, createdEvents);
+      datastore.put(entity);
+  }
+
+  private GoogleIdToken verifyId(String idTokenString) {
```

Comment:
refactor this and UserServlet.java:198 into GoogleIdHelper.java so that both EventsServlet and UsersServlet could use code reading something like:

`Payload p = GoogleIdHelper.verifyId(request)`

The following pattern is seen three times in UserServlet, and twice here - a sure sign it could be in a helper somewhere!
```
GoogleIdToken idToken = verifyId(request.getParameter(ID_TOKEN_PARAM));
    if (idToken == null) {
        System.out.println("Invalid ID token.");
        response.setStatus(400);
        return;
    }

    Payload payload = idToken.getPayload();
```

What the code essentially does is converts the ID_TOKEN request parameter into a usable Payload. This refactor can also make it easier to test by keeping the dependency injection stuff compartmentalized to the GoogleIdHelper.